### ISSUE TITLE
Flow on ember-cli apps

### ISSUE THREAD

**Issue by Globegitter** [facebook/flow#287]
I am just trying out flow 0.4.0 (`flow check`) on ember-cli apps and just testing on one file for now and getting the following error `project/app/app.js:
Failure("Unimplemented: ImportDeclaration")`

The file is the following:
```js
/* @flow weak */
import Ember from 'ember';
import Resolver from 'ember/resolver';
import loadInitializers from 'ember/load-initializers';
import config from './config/environment';

Ember.MODEL_FACTORY_INJECTIONS = true;

var App = Ember.Application.extend({
  modulePrefix: config.modulePrefix,
  podModulePrefix: config.podModulePrefix,
  Resolver: Resolver
});

loadInitializers(App, config.modulePrefix);

export default App;
```

ember-cli is using ES6 imports, see here: http://www.ember-cli.com/#using-modules

I tried to declare an interface file with modules and classes, but it still throws the same error Any idea what the issue could be?

I am more than happy to provide any more information that would help resolve this error.

**Comment by jeffmo** [comment]
The bad new is that Flow doesn't understand ES6 modules yet.

The good news is that I've just made some really good progress on ES6 module support this week; So hopefully this should be ready very soon. (I'll over-estimate a bit and say it should be ready *and* stable in the next month or so)

**Comment by Globegitter** [comment]
Ahh that is great news - thanks. More out of curiosity: The build pipeline is outputting AMD modules. So in theory at least, if piped in the right way one could run it on the outputted file, right?

**Comment by jeffmo** [comment]
tldr: Yes, that should work.

The plan is to get analysis working correctly first -- which means flow can understand the code and how imports/exports link together. The second stage (which is downstream from that) would be to have a canonical reference implementation of an es6 module transpiler in the `jsx` tool. Building a transpiler is not at all tied to the analysis portion, though, so it's not as high priority since there already exist some pretty good ES6 module transpilers out there.

In other words: You should be able to run any spec-compliant ES6 module code through flow to get error information (much like you'd use a linter). For transpiling, you can use ember-cli, Babel, es6-transpiler, or (eventually) the `jsx` tool, etc -- whichever fits your needs in terms of toolchain or output.

**Comment by jussi-kalliokoski** [comment]
:+1: for adding ES6 module syntax support

**Comment by jeffmo** [comment]
@jussi-kalliokoski : I've been bad at following up on issues, but import/export support shipped in flow v0.7. Give it a shot!

It should work fairly well with Babel's ES6 module transpilation strategy, but TBH we haven't dogfooded this extensively yet -- so please let me know if you run into any issues.

**Comment by jussi-kalliokoski** [comment]
@jeffmo: thanks, that's great and seems to work! although now I tripped up on `let` and `const` not being supported. x)
